tests/unit/test_optional_ids.py

```diff
@@ -0,0 +1,175 @@
+from power_grid_model.validation import assert_valid_batch_data, assert_valid_input_data
+
+
+@pytest.fixture
+def input_dataset_row():
+    node = initialize_array(DatasetType.input, ComponentType.node, 3)
+    node["id"] = np.array([1, 2, 6])
+    node["u_rated"] = [10.5e3, 10.5e3, 10.5e3]
+
+    line = initialize_array(DatasetType.input, ComponentType.line, 3)
+    line["id"] = [3, 5, 8]
+    line["from_node"] = [1, 2, 1]
+    line["to_node"] = [2, 6, 6]
+    line["from_status"] = [1, 1, 1]
+    line["to_status"] = [1, 1, 1]
+    line["r1"] = [0.25, 0.25, 0.25]
+    line["x1"] = [0.2, 0.2, 0.2]
+    line["c1"] = [10e-6, 10e-6, 10e-6]
+    line["tan1"] = [0.0, 0.0, 0.0]
+    line["i_n"] = [1000, 1000, 1000]
+
+    sym_load = initialize_array(DatasetType.input, ComponentType.sym_load, 2)
+    sym_load["id"] = [4, 7]
+    sym_load["node"] = [2, 6]
+    sym_load["status"] = [1, 1]
+    sym_load["type"] = [LoadGenType.const_power, LoadGenType.const_power]
+    sym_load["p_specified"] = [20e6, 10e6]
+    sym_load["q_specified"] = [5e6, 2e6]
+
+    source = initialize_array(DatasetType.input, ComponentType.source, 1)
+    source["id"] = [10]
+    source["node"] = [1]
+    source["status"] = [1]
+    source["u_ref"] = [1.0]
+
+    return {
+        ComponentType.node: node,
+        ComponentType.line: line,
+        ComponentType.sym_load: sym_load,
+        ComponentType.source: source,
+    }
+
+
+@pytest.fixture
+def input_dataset_col(input_dataset_row):
+    return compatibility_convert_row_columnar_dataset(
+        input_dataset_row, ComponentAttributeFilterOptions.relevant, DatasetType.input
+    )
+
+
+@pytest.fixture(params=["input_dataset_row", "input_dataset_col"])
+def input_dataset(request):
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture
+def batch_update_sym_load_row():
+    sym_load = initialize_array(DatasetType.update, ComponentType.sym_load, (2, 2))
+    sym_load["id"] = [[4], [7]]
+    sym_load["p_specified"] = [[30e6], [15e6]]
+    return sym_load
+
+
+@pytest.fixture
+def batch_update_sym_load_no_id_row():
+    sym_load = initialize_array(DatasetType.update, ComponentType.sym_load, (2, 2))
+    sym_load["p_specified"] = [[30e6, 10e6], [30e6, 15e6]]
+    return sym_load
+
+
+@pytest.fixture
+def batch_update_sym_load_col():
+    source_attribute_dtypes = power_grid_meta_data[DatasetType.update][ComponentType.sym_load].dtype
+    return {
+        "id": np.array([[4], [7]], dtype=source_attribute_dtypes["id"]),
+        "p_specified": np.array([[30e6], [15e6]], dtype=source_attribute_dtypes["p_specified"]),
+    }
+
+
+@pytest.fixture
+def batch_update_sym_load_no_id_col():
+    source_attribute_dtypes = power_grid_meta_data[DatasetType.update][ComponentType.sym_load].dtype
+    return {
+        "p_specified": np.array([[30e6, 10e6], [30e6, 15e6]], dtype=source_attribute_dtypes["p_specified"]),
+    }
+
+
+@pytest.fixture(
+    params=[
+        "batch_update_sym_load_row",
```

Comment:
```suggestion
        "batch_update_sym_load_row",
```
Can you use really short names? They would appear bigger  in the sidebar.
If function names look ugly, you can also use `@pytest.fixture(name= ...)` to rename fixtures instead of functions but maybe it wont be helpful.

maybe we can also use a symbol instead full name.

Reply:
[ed9c9b9](https://github.com/PowerGridModel/power-grid-model/pull/819/commits/ed9c9b91a1cad074a12073ffaf9614ee937f97ae) should be better